### ISSUE TITLE
Test suite should not run servers on public interfaces

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/opa#6162]
Running `make test` on my Mac now paints a cascade of windows where each warn about a test binding to a public interface. This is apparently the firewall in Mac OS being helpful, and it's probably best to keep it enabled. We should make sure the tests we run don't have servers binding to public interfaces unless needed for some reason.

![thisisfine](https://github.com/open-policy-agent/opa/assets/510711/5591fc08-709d-4ac2-b5cb-978cbfb7f9b7)

**Comment by charlieegan3** [comment]
➕ thanks for filing, I am also seeing this.

**Comment by johanfylling** [comment]
I don't think you actually need to select "allow" on any of the dialogs for the tests to pass, so localhost connections are probably let through; but this is still very annoying, so we should definitely fix it 👍 .
